bokeh/events.py

```diff
@@ -149,14 +152,14 @@ def __init_subclass__(cls):
             _CONCRETE_EVENT_CLASSES[cls.event_name] = cls
 
     @classmethod
-    def decode_json(cls, dct: EventJson) -> Event:
+    def decode_json(cls, rep: EventRep, decoder: Deserializer) -> Event:
         ''' Custom JSON decoder for Events.
 
         Can be used as the ``object_hook`` argument of ``json.load`` or
         ``json.loads``.
 
         Args:
-            dct (dict) : a JSON dictionary to decode
+            rep (dict) : a JSON dictionary to decode
```

Comment:
This docstring needs a complete rewrite, as it must not refer to JSON in this context. I use "rep" as a shorthand for "(serializable) representation". I would use "repr", but that would be confusing with `repr()`. I need to stress that the new serialization scheme does not produce JSON, it produces serializable (or intermediate) representation of objects, that can be later encoded as a JSON string or something else in future, and possibly it could be skipped altogether if had some form of direct serialization, e.g. if we had a binary serializer. This representation is geared towards allowing to pass fairly arbitrary objects to JS and back, where JSON is just an implementation detail as a wire encoding.